libobs/audio-monitoring/win32/wasapi-output.c

```diff
@@ -451,11 +451,11 @@ void audio_monitor_destroy(struct audio_monitor *monitor)
 {
 	if (monitor) {
 		audio_monitor_free(monitor);
-
-		pthread_mutex_lock(&obs->audio.monitoring_mutex);
-		da_erase_item(obs->audio.monitors, &monitor);
-		pthread_mutex_unlock(&obs->audio.monitoring_mutex);
-
+		if (obs->audio.monitors.num) {
```

Comment:
Shouldn't this also be protected by the mutex?

Reply:
Which mutex should this be protected by? The obs->audio.monitoring_mutex can be destroyed at this point, that was the point of this pull request.

Reply:
The mutex presumably protects `obs->audio.monitors`, if it's being destroyed before operations affecting `obs->audio.monitors` have finished then that seems like the bug that should be fixed.

Reply:
if obs->audio.monitors.num is 0 than obs->audio.monitors is not used, so the obs->audio.monitoring_mutex is not needed at that point.
There can go nothing wrong if obs->audio.monitors.num is changed during the checking of it.